### DISCUSSION TITLE
basic-grammar/type-cast

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#97] [Announcements]
# basic-grammar/type-cast

Java程序员进阶之路，小白的零基础Java教程，Java 数据类型转换（强制数据类型转换+自动类型转换）

https://tobebetterjavaer.com/basic-grammar/type-cast.html

**Comment by DylanLJC** [comment]
一共付给收银员44.99999923706055元
这里为什么是这个数需要解释一下？

个人认为：应该是先将浮点型转为二进制，与二进制int型相乘得到的吧

**Reply by itwanger** [reply]
这个例子就是简单说一下类型转换，哈哈

**Comment by BOUNDLESS-558** [comment]
虽然我已经不是零基础了，但作者这篇写的还是不错的。

**Comment by NobodyUvU** [comment]
“自动类型转换并不局限于小类型到大类型，也可以是大类型到小类型，只要值在小类型的取值范围内，编译器就会自动帮我们做隐式转换。”，这句话应该只适用于字面量，不适用于变量吧？如果我的理解没错的话
